rust/ommx/src/instance.rs

```diff
@@ -50,3 +50,285 @@ pub struct Instance {
     #[getset(get = "pub")]
     constraint_hints: ConstraintHints,
 }
+
+impl Instance {
+    ///
+    /// - All `VariableID`s in `Function`s contained both directly and indirectly must be keys of `decision_variables`.
+    /// - Key of `constraints` and `removed_constraints` are disjoint.
+    /// - The keys of `decision_variable_dependency` are also keys of `decision_variables`.
+    ///
+    #[allow(clippy::too_many_arguments)]
+    pub fn try_new(
+        sense: Sense,
+        objective: Function,
+        decision_variables: BTreeMap<VariableID, DecisionVariable>,
+        constraints: BTreeMap<ConstraintID, Constraint>,
+        removed_constraints: BTreeMap<ConstraintID, RemovedConstraint>,
+        decision_variable_dependency: BTreeMap<VariableID, Function>,
+        parameters: Option<v1::Parameters>,
```

Comment:
Arbitrary `parameters` are allowed. This should be public member, and do not include `new` arugment